### DISCUSSION TITLE
Is whatsapp component supports template message?

### DISCUSSION THREAD

**Discussion by VelSankar-DT** [apache/camel-karavan#1064] [Q&A] [answered]
I want to send a template message to WhatsApp chat through our Camel Karavan WhatsApp component. Like the template format is, In the header section I want to send a document, and in the body section I want to send text content.

Is that possible to do? If yes can I get any reference document?

**Comment by davsclaus** [comment]
Its not easy, but look in the source code for camel-whatsapp, in the model package. You need to find what whatapp message type that works the best for you, and then build those model java classes with your content. And set that as the message body, such as a `InteractiveMessageRequest`.

**Comment by davsclaus** [comment, marked as answer]
Also you can likely build such request from a JSon structured document and have it converted into the `xxxRequest` java class that camel-whatsapp uses. However maybe we can improve camel-whatsapp in the future to accept json as input
